controllers/admin/AdminCartRulesController.php

```diff
@@ -201,9 +201,43 @@ public function postProcess()
             }
 
             // These are checkboxes (which aren't sent through POST when they are not checked), so they are forced to 0
+            $restrictions_selected = false;
             foreach (['country', 'carrier', 'group', 'cart_rule', 'product', 'shop'] as $type) {
                 if (!Tools::getValue($type . '_restriction')) {
                     $_POST[$type . '_restriction'] = 0;
+                } elseif ($type !== 'shop') { // shop restriction is forced to be checked, thus we do not take it into account
+                    $restrictions_selected = true;
+                }
+            }
+
+            // At least one of these is compulsory: code, customer or any restriction. If none of these are present, the cart rule applies to all carts !
+            if (empty(Tools::getValue('id_cart_rule')) &&
+                !$restrictions_selected &&
+                empty(Tools::getValue('id_customer')) && empty(Tools::getValue('code')) && empty(Tools::getValue('minimum_amount'))
+                && Tools::getValue('date_from') < date('Y-m-d H:i:s')
+            ) {
+                $this->errors[] = $this->trans('The starting time is in the past. To avoid any issue, create a promo code, select a condition or set a starting time in the future.', [], 'Admin.Catalog.Notification');
+            }
+
+            // If the restriction is checked, but no item is selected, raise an error
+            foreach (['country', 'carrier', 'group', 'shop'] as $type) {
+                if (Tools::getValue($type . '_restriction') && empty(Tools::getValue($type . '_select'))) {
+                    switch ($type) {
```

Comment:
Isn't a map better than a switch case for this usage ? :thinking: 
The only thing i see easier is default behavior but I think it would be easy to solve with a ternary operator :thinking: 


What do you think ^^